cli/command/trust/signer-remove.go

```diff
@@ -0,0 +1,123 @@
+package trust
+
+import (
+	"fmt"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/spf13/cobra"
+)
+
+type signerRemoveOptions struct {
+	forceYes bool
+	images   []string
+	signer   string
+}
+
+func newSignerRemoveCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-remove IMAGE NAME",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.signer = args[0]
+			options.images = args[1:]
+			return removeSigner(dockerCli, &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, options *signerRemoveOptions) error {
+	if len(options.images) < 1 {
+		return fmt.Errorf("at least one signer must be provided")
+	}
+
+	for _, image := range options.images {
+		if err := removeSingleSigner(cli, image, options.signer, options.forceYes); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isLastSigner(role data.Role, notaryRepo *client.NotaryRepository) (bool, error) {
+	allRoles, err := notaryRepo.ListRoles()
+	if err != nil {
+		return true, err
+	}
+	var lastSigner client.RoleWithSignatures
+	for i := range allRoles {
+		if allRoles[i].Name == "targets/releases" {
+			lastSigner = allRoles[i]
+		}
+	}
+	if len(lastSigner.Signatures) == 0 {
+		fmt.Printf("All tags are currently unpullable, use docker sign to fix")
+		return false, nil
+	}
+	return (role.KeyIDs[0] == lastSigner.Signatures[0].KeyID), nil
```

Comment:
While it may not be possible to produce this metadata with a notary client today, I think we should do a double for loop over role key ids and signature key ids.

Also, it will prevent us from indexing into a potentially empty slice.